### DISCUSSION TITLE
Is it possible to upgrade JJ to 4.x without upgrading JJ+?

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#2652] [Q&A] [answered]
### Introduction

Hi!

First of all, kudos for the 4.x release!

I'm currently on the latest 3.x release of both JJ and JJ+ and I'm wondering if I can upgrade JJ to 4.x without upgrading JJ+ to 4.x (since there is no 4.x for the moment being). Would that work? Or do I need to wait for the JJ+ 4.0 release?

Regards!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
> (since there is no 4.x for the moment being)

We are currently on *JointJS+* `v4.0.1`.

---

It is possible to use *JointJS+* and *JointJS* if they differ in patch version.

If a new patch version of *JointJS* is released, you can delete your `package-lock.json`  (or `yarn.lock`) and install *JointJS+* again. It will pull the latest *JointJS* as we allow patches in *JointJS+* `package.json` (using `~`).

**Reply by alexandernst** [reply]
Well... thats embarrassing 😳... I saw the 4.x release and thought it was only JJ. I do recall reading the release notes and thinking "I'm gonna need to do some refactoring", but somehow my brain totally ignored the fact that JJ+ was also released. Anyways, thank you Roman! 🙏

**Reply by kumilingus** [reply]
No worries, I suffer from the same condition 😄
